microsetta_public_api/api/microsetta_public_api.yml

```diff
@@ -411,18 +601,162 @@ components:
         type: string
         example: "greengenes"
       required: true
+    percentiles:
+      name: percentiles
+      in: query
+      explode: false
+      schema:
+        $ref: '#/components/schemas/percentiles'
+      required: false
 
   schemas:
     alphaMetric:
       type: string
       example: "faith_pd"
+    anyValue:
+      nullable: true
+    betaMetric:
+      type: string
+      example: "unifrac"
+    coordinates:
+      description: >
+        2D array of coordinates
+      type: array
+      items:
+        type: array
+        items:
+          type: number
+      example:
+        - [-0.1776, 0.6011, -0.2033, -0.3368]
+        - [0.2561, 0.4013, -0.32037, 0.09316]
+    emperorDecomposition:
+      type: object
+      properties:
+        coordinates:
+          $ref: '#/components/schemas/coordinates'
+        percents_explained:
+          $ref: '#/components/schemas/percentsExplained'
+        sample_ids:
+          $ref: '#/components/schemas/sampleIdArray'
+    emperorPCoA:
+      type: object
+      description: >
+        # TODO
+      properties:
+        decomposition:
+          $ref: '#/components/schemas/emperorDecomposition'
+        metadata:
+          $ref: '#/components/schemas/metadata'
+        metadata_headers:
+          $ref: '#/components/schemas/metadataHeaders'
+    metadata:
+      description: >
+        2d array of metadata values
+      type: array
+      items:
+        type: array
+        items:
+          $ref: '#/components/schemas/metadataValue'
+        example:
+          - ["fecal", 0.7, 9, "30s", "Normal"]
+          - ["sebum", 20.4, null, "40s", "Overweight"]
+    metadataHeaders:
+      description: >
+        List of metadata column names
+      type: array
+      items:
+        type: string
+      example:
+        - "body-habitat"
+        - "latitude"
+        - "days_post_surgery"
+        - "age_cat"
+        - "bmi_cat"
+    metadataQuery:
+      # TODO try to make this a little more informative (challenging due to recursive definition)
```

Comment:
Cool. Hadn't tried recursive schema yet. Glad it looks like it works, even if it might not render well in the UI.

Happy to leave the informative example with permissive schema for now and revisit later.